gateway-service-impl/src/main/java/org/hypertrace/gateway/service/baseline/BaselineServiceQueryParser.java

```diff
@@ -0,0 +1,229 @@
+            .addChildFilter(createTimeFilter(timeColumn, Operator.LT, endTimeInMillis)));
+
+    return filterBuilder;
+  }
+
+  private static Filter createTimeFilter(String columnName, Operator op, long value) {
+    ColumnIdentifier.Builder timeColumn = ColumnIdentifier.newBuilder().setColumnName(columnName);
+    org.hypertrace.core.query.service.api.Expression.Builder lhs =
+        org.hypertrace.core.query.service.api.Expression.newBuilder()
+            .setColumnIdentifier(timeColumn);
+    LiteralConstant.Builder constant =
+        LiteralConstant.newBuilder()
+            .setValue(
+                org.hypertrace.core.query.service.api.Value.newBuilder()
+                    .setValueType(org.hypertrace.core.query.service.api.ValueType.LONG)
+                    .setLong(value));
+    org.hypertrace.core.query.service.api.Expression.Builder rhs =
+        org.hypertrace.core.query.service.api.Expression.newBuilder().setLiteral(constant);
+    return Filter.newBuilder().setLhs(lhs).setOperator(op).setRhs(rhs).build();
+  }
+
+  public BaselineEntitiesResponse parseQueryResponse(
+      Iterator<ResultSetChunk> resultSetChunkIterator,
+      BaselineRequestContext requestContext,
+      int idColumnsSize,
+      String entityType,
+      long periodSecs) {
+    Map<String, AttributeMetadata> attributeMetadataMap =
+        attributeMetadataProvider.getAttributesMetadata(requestContext, entityType);
+    Map<EntityKey, Map<String, BaselineMetricSeries.Builder>> entityMetricSeriesMap =
+        new LinkedHashMap<>();
+    while (resultSetChunkIterator.hasNext()) {
+      ResultSetChunk chunk = resultSetChunkIterator.next();
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Received chunk: " + chunk.toString());
+      }
+
+      if (chunk.getRowCount() < 1) {
+        break;
+      }
+
+      for (Row row : chunk.getRowList()) {
+        EntityKey entityKey =
+            EntityKey.of(
+                IntStream.range(0, idColumnsSize)
+                    .mapToObj(value -> row.getColumn(value).getString())
+                    .toArray(String[]::new));
+
+        Map<String, BaselineMetricSeries.Builder> metricSeriesMap =
+            entityMetricSeriesMap.computeIfAbsent(entityKey, k -> new LinkedHashMap<>());
+
+        BaselineInterval.Builder intervalBuilder = BaselineInterval.newBuilder();
+
+        // Second column is the time column
+        Value value =
+            QueryAndGatewayDtoConverter.convertQueryValueToGatewayValue(
+                row.getColumn(idColumnsSize));
+        if (value.getValueType() == ValueType.STRING) {
+          long time = Long.parseLong(value.getString());
+          intervalBuilder.setStartTimeMillis(time);
+          intervalBuilder.setEndTimeMillis(time + TimeUnit.SECONDS.toMillis(periodSecs));
+
+          for (int i = idColumnsSize + 1;
+              i < chunk.getResultSetMetadata().getColumnMetadataCount();
```

Comment:
Nope, it is available for all chunks

Reply:
It's not:
https://github.com/hypertrace/query-service/blob/main/query-service-api/src/main/proto/response.proto#L25-L26 (implementation confirms this comment)

Reply:
I have verified locally it is available for all chunks and same code is running in prod as well.

Reply:
Same logic exists everywhere and its already tested and it is working as expected. Not sure why it is like that. It will be fixed as part of https://traceableai.atlassian.net/browse/ENG-7467

Reply:
I'm confused, because I know this behavior is still around. I reimplemented this myself fairly recently, and left the behavior as it was to maintain compatibility - https://github.com/hypertrace/query-service/blob/main/query-service-impl/src/main/java/org/hypertrace/core/query/service/RowChunkingOperator.java Are you sure you're requesting enough data to actually get multiple chunks?

Anyway, isn't it easy to fix regardless?

Reply:
Fixed this as well.